web/annotation_layer_builder.js

```diff
@@ -195,6 +215,37 @@ class AnnotationLayerBuilder {
     return !!this.annotationLayer?.hasEditableAnnotations();
   }
 
+  /**
+   * @param {InjectLinkAnnotationsOptions} options
+   * @returns {Promise<void>} A promise that is resolved when the inferred links
+   *   are added to the annotation layer.
+   */
+  async injectLinkAnnotations({
+    inferredLinks,
+    viewport,
+    structTreeLayer = null,
+  }) {
```

Comment:
We should also ensure that `render` has completed *before* this method is invoked, which should be achievable the following way:

 - Apply the following diff.
```diff
diff --git a/web/annotation_layer_builder.js b/web/annotation_layer_builder.js
index b6e31404c..90ae789b3 100644
--- a/web/annotation_layer_builder.js
+++ b/web/annotation_layer_builder.js
@@ -133,6 +133,8 @@ class AnnotationLayerBuilder {
     this.#onAppend?.(div);

     if (annotations.length === 0) {
+      this.#annotations = annotations;
+
       this.hide();
       return;
     }
```

 - Add the following check in this method.
```suggestion
  }) {
    if (this.#annotations === null) {
      throw new Error("`render` must have been called.");
    }
```